### PR TITLE
effective config doesn't include advanced.config mods

### PR DESCRIPTION
**Pull request by joedevivo**
This is a tricky one, I'd like to get one of the following solutions in for riak's 2.0 release:
### Option 1

simply cat the advanced.config at the end of the output of the effective conf. It will at least indicate to the user that these values are detected and being used.
### Option 2

Since we know which app.config values a cuttlefish variable will effect, we could include some kind of indicator that it is overridden in advanced.config.

e.g.

```
some.setting.a = on
## some.setting.b is overriden in advanced.config's 'erlang_library.variable'
## some.setting.b = off
```

I think this option is more friendly to commandline querying of effective confs. grep etc...

### CODE REVIEW

**Comment by bookshelfdave** [comment]
all eunit tests pass: https://gist.github.com/metadave/9896753

File: src/cuttlefish_effective.erl [hunk 1]
```diff
@@ -0,0 +1,246 @@
+%% -------------------------------------------------------------------
+%%
+%% cuttlefish_effective: handles generating the effective configuration
+%%
+%% Copyright (c) 2013 Basho Technologies, Inc.  All Rights Reserved.
```
Comment (bookshelfdave) [inline src/cuttlefish_effective.erl]: 2014

File: src/cuttlefish_effective.erl [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                    [?FMT("## ~s was overridden in advanced.config", [Variable]),
+                     "## " ++ Line] ++ Acc;
+                _ -> [Line | Acc]
+            end
+        end,
+        [],
+        EffectiveConfig
+    ).
+
+%% @doc check's a mapping's "mapping"
```
Comment (bookshelfdave) [inline src/cuttlefish_effective.erl]: check _what_ for a mapping's mapping?

**Comment by bookshelfdave** [comment]
it looks like advanced.config vars that don't have a cuttlefish equiv (such as repl) won't show up in the output.

**Comment by bookshelfdave** [comment]
latest eunit output: https://gist.github.com/metadave/9901069

**Comment by bookshelfdave** [comment]
+1 52e963d

**Comment by joedevivo** [comment]
@borshop merge

**Comment by cmeiklejohn** [comment]
Super pedantic, but in the future I'd really like to see commit histories like this squash so there aren't a bunch of broken WIP intermediary commits in the history.

**Comment by jrwest** [comment]
+1 to that @cmeiklejohn

**Pull request merged**